### PR TITLE
feat(public-cloud): instance creation

### PR DESCRIPTION
**Pull request by sachinrameshn**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/pci-localzone` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-12754 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [x] Breaking change is mentioned in relevant commits

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/instances/add/add.controller.js [hunk 1]
```diff
@@ -416,19 +429,36 @@ export default class PciInstancesAddController {
...
-    if (this.model.datacenter.type === this.LOCAL_ZONE_REGION) {
-      this.PciProjectsProjectInstanceService.getLocalPrivateNetworks(
-        this.projectId,
-        this.model.datacenter.name,
+  getLocalPrivateNetworkSubnet() {
+    return this.$q
+      .all(
+        this.availableLocalPrivateNetworks.map((privateNetwork) => {
+          if (privateNetwork.id) {
+            this.PciProjectsProjectInstanceService.getLocalPrivateNetworkSubnets(
```
Comment (qpavy) [inline packages/manager/modules/pci/src/projects/project/instances/add/add.controller.js:437]: Don't you have to return the result of this Promise as well ?
```suggestion
            return this.PciProjectsProjectInstanceService.getLocalPrivateNetworkSubnets(
```

File: packages/manager/modules/pci/src/projects/project/instances/instance/edit/edit.routing.js [hunk 2]
```diff
@@ -60,7 +60,7 @@ export default /* @ngInject */ ($stateProvider) => {
       },
       imageInformation: /* @ngInject */ ($http, projectId, instance) => {
         return $http
-          .get(`/cloud/project/${projectId}/image/${instance.image.id}`)
+          .get(`/cloud/project/${projectId}/image/${instance.image?.id}`)
```
Comment (ghyenne) [inline packages/manager/modules/pci/src/projects/project/instances/instance/edit/edit.routing.js:63]: So if image is undifined you call .get(`/cloud/project/${projectId}/image/undefined`) i think this case shoudn't exist.

File: packages/manager/modules/pci/src/projects/project/instances/add/add.controller.js (hunk 1 above)
Comment (sachinrameshn) [reply packages/manager/modules/pci/src/projects/project/instances/add/add.controller.js:437]: yup Thanks

File: packages/manager/modules/pci/src/components/project/flavors-list/flavors-list.service.js [hunk 3]
```diff
@@ -129,27 +129,15 @@ export default class FlavorsList {
...
-    this.locationCompatibility = {
-      isLocalZone: false,
-      isGlobalZone: true,
+  getlocationCompatibility(productAvailability) {
+    return {
+      isLocalZone: productAvailability?.regions.some(
+        (region) => region.type === this.LOCAL_ZONE,
+      ),
+      isGlobalZone: productAvailability?.regions.some(
+        (region) => region.type !== LOCAL_ZONE,
```
[the remark this answers is not in the document]
Comment (sachinrameshn) [reply packages/manager/modules/pci/src/components/project/flavors-list/flavors-list.service.js:138]: yeah my bad I missed it, I have added it to resolve Lint Error.

File: packages/manager/modules/pci/src/projects/project/instances/instance/edit/edit.routing.js (hunk 2 above)
Comment (sachinrameshn) [reply packages/manager/modules/pci/src/projects/project/instances/instance/edit/edit.routing.js:63]: Sometimes instance dont have images for few instances, I will add a if condition to resolve this issue.

**Review by qpavy — APPROVED** [review]

**Review by ghyenne — APPROVED** [review]

**Pull request merged**